### DISCUSSION TITLE
How to handle the comments string?

### DISCUSSION THREAD

**Discussion by evildrome** [kevinzg/facebook-scraper#650] [Q&A]
Sorry, bit of a noob question... what format is the comments string in?  Is it nested dictionaries ?

It kinda looks a bit like JSON...

I want to read it in & iterate over it to get the comment_id, commenter_id, comment_text, etc  one after the other to populate an SQLITE database to be consumed by my FB data viewer program.

This is a demo of the Viewer.

https://www.youtube.com/watch?v=N37EyJwgRhI

TIA!

Wilson.

**Comment by neon-ninja** [comment]
See https://github.com/kevinzg/facebook-scraper/issues/303#issuecomment-856394985 for an example.

It's a list of dictionaries. If there's replies, then those are included as a list of dictionaries within the comment, under the `replies` key. Python dictionaries look almost like JSON, but they're not valid JSON. You can convert them to JSON easily though, with `json.dump` or `json.dumps`. If you set the `default=str` parameter, `json.dumps` will even convert the datetime objects to strings for you. You can use the Python standard library sqlite3 for interfacing with sqlite database files - https://docs.python.org/3/library/sqlite3.html

**Reply by evildrome** [reply]
Thanks very much!
